tracing-opentelemetry/src/main/java/com/couchbase/client/tracing/opentelemetry/OpenTelemetryRequestTracer.java

```diff
@@ -68,19 +69,17 @@ private Span castSpan(final RequestSpan requestSpan) {
    * Returns the inner OpenTelemetry tracer.
    */
   public Tracer tracer() {
-    return  tracer;
+    return tracer;
   }
 
   @Override
   public RequestSpan requestSpan(String operationName, RequestSpan parent) {
     SpanBuilder spanBuilder = tracer.spanBuilder(operationName);
+    Context parentContext = Context.current();
     if (parent != null) {
-      spanBuilder.setParent(Context.current().with(castSpan(parent)));
-    } else {
-      spanBuilder.setNoParent();
```

Comment:
actually one more question here too: I assume you are refering to the implicit propagation through active and thread locals? What about where we are working under async / reactive environments where the one in the thread local might not actually be the "real" parent?

Btw, not sure if you noticed, we actually have an explicit propagation mechanism. On every request the user makes, in the option block, you can provide the parent explicitly.

Reply:
Yeah for async environments, we still expect the context to be propagated through callbacks, for example wiring up using something like 

```
Executor contextExecutor = Context.current().wrap(Runnable::run);
asyncClient.execute(request).thenComposeAsync(response -> asyncClient.execute(request2), contextExecutor)
...;
```

Or registering the executor in a dagger graph, or as a reactive subscriber, or maybe use reactive hooks via some instrumentation like https://github.com/open-telemetry/opentelemetry-java-instrumentation/tree/main/instrumentation/reactor-3.1/library, or using the java agent which also does automatic context propagation in many cases.

The reality is very few libraries support explicitly passing in a context like this couchbase one does, so propagating the context into threadlocals across async bounds is basically a requirement in these environments, e.g. if you're using gRPC or really any other library you need to be propagating context. On the bright side, this means that a couchbase user doesn't have to worry about couchbase specifically, they can treat it as any other client in their app and expect it to use the implicit parent when there's no explicit one.

Of course, it's up to you whether this is something you want to support - on the flip side, you could just have a couchbase sdk policy saying "always pass in the parent, we don't use implicit propagation at all". I wouldn't recommend it though since it'd be a jarring experience for users compared to other libraries.